src/lib/openapi.ts

```diff
@@ -1,82 +1,132 @@
-  operationId: string
-  summary?: string
-  description?: string
-  parameters?: OpenapiParameter[]
-  requestBody?: OpenapiRequestBody
-  responses: Record<string, OpenapiResponse>
-  tags?: string[]
-  security?: OpenapiSecurity[]
-}
-
-export interface OpenapiParameter {
-  name: string
-  in: 'query' | 'header' | 'path' | 'cookie'
-  description?: string
-  required?: boolean
-  schema: OpenapiSchema
-}
-
-export interface OpenapiRequestBody {
-  content: Record<string, OpenapiMediaType>
-  description?: string
-  required?: boolean
-}
-
-export interface OpenapiResponse {
-  description: string
-  content?: Record<string, OpenapiMediaType>
-}
-
-export interface OpenapiMediaType {
-  schema: OpenapiSchema
-}
-
-export interface OpenapiSchema {
-  type: 'object' | 'array' | 'string' | 'number' | 'integer' | 'boolean'
-  properties?: Record<string, OpenapiSchema>
-  items?: OpenapiSchema
-  $ref?: string
-  required?: string[]
-  format?: string
-  description?: string
-}
-
-export interface OpenapiComponents {
-  schemas: Record<string, OpenapiSchema>
-}
-
-export type OpenapiSecurity = Record<string, string[]>
+import { z } from 'zod'
+
+const ZStringWithFormat = z.object({
+  type: z.literal('string'),
+  format: z.string().optional(),
+  description: z.string().optional(),
+})
+
+const ZNumber = z.object({
+  type: z.enum(['number', 'integer']),
+  format: z.string().optional(),
+  description: z.string().optional(),
+})
+
+const ZBoolean = z.object({
+  type: z.literal('boolean'),
+  description: z.string().optional(),
+})
+
+const ZSchema: z.ZodType<any> = z.lazy(() =>
+  z.union([
+    z.object({
+      type: z.literal('object'),
+      properties: z.record(ZSchema).optional(),
+      required: z.array(z.string()).optional(),
+      description: z.string().optional(),
+    }),
+    z.object({
+      type: z.literal('array'),
+      items: ZSchema,
+      description: z.string().optional(),
+    }),
+    ZStringWithFormat,
+    ZNumber,
+    ZBoolean,
+    z.object({
+      $ref: z.string(),
+    }),
+  ]),
+)
+
+export const ZOpenapiInfo = z.object({
+  title: z.string(),
+  version: z.string(),
+})
+
+export const ZOpenapiServer = z.object({
+  url: z.string(),
+})
+
+export const ZOpenapiTag = z.object({
+  name: z.string(),
+  description: z.string(),
+})
+
+export const ZOpenapiParameter = z.object({
+  name: z.string(),
+  in: z.enum(['query', 'header', 'path', 'cookie']),
+  description: z.string().optional(),
+  required: z.boolean().optional(),
+  schema: ZSchema,
+})
+
+export const ZOpenapiRequestBody = z.object({
+  content: z.record(
+    z.object({
+      schema: ZSchema,
+    }),
+  ),
+  description: z.string().optional(),
+  required: z.boolean().optional(),
+})
+
+export const ZOpenapiResponse = z.object({
+  description: z.string(),
+  content: z
+    .record(
+      z.object({
+        schema: ZSchema,
+      }),
+    )
+    .optional(),
+})
+
+export const ZOpenapiOperation = z.object({
```

Comment:
The `Z` prefix is painful. Should we just set this eslint rule `@typescript-eslint/no-redeclare` to `ignoreDeclarationMerge` so we can name the schema and the type the same per the Zod docs? We don't have this issue in most other repos, but things would be much simpler if we allowed it here. 

I have the same issue in my other PR, going to try it there.

Reply:
Ok, nevermind, using the same name doesn't work. Instead of the Z prefix, how about `Schema` suffix?

Reply:
I've followed this convention in https://github.com/seamapi/blueprint/pull/26

Reply:
I wanted to avoid the atrocity of `schemaSchema` but maybe something like `modelSchema`?

Reply:
Just wanted to note that while having one instance of `OpenapiSchemaSchema` is not ideal, it shouldn't mean the convention is no good.